* A Transfer Learning Framework for Organelle Proteomics Data

LM Breckels^{12}, S Holden^{3}, KS Lilley^{2}, L Gatto^{12}

^{1} Computational Proteomics Unit, ^{2} Cambridge Centre for Proteomics, ^{3} The Computer Laboratory, University of Cambridge, UK

** Introduction and objectives 

Organelle proteomics, or spatial proteomics, is the systematic study
of proteins and their assignment to organelles, and is a field of
rapidly growing importance. The knowledge of subcellular localisation
of proteins is extremely desirable to biologists, as it can assist
elucidation of a protein's role within the cell, as proteins are
spatially organised according to their function and specificity of
their molecular interactions. Experimental organelle proteomics
requires sophisticated experimental designs and data analyses in order
to obtain accurate datasets, and to collectively infer the
localisation of thousands of proteins [1]. Here, we use 
freely available data from repositories to improve upon the 
classification of experimental, condition and organism-specific 
protein subcellular localistation predictions. 

** Methods

We use state-of-the-art machine learning in a transfer learning
framework [2] to simultaneously exploit several sources of information
available with which to assign a protein to its subcellular
compartment. These sources encompass our main (primary) data produced
from experimental high-throughput mass spectrometry (MS)-based methods
and _in silico_ (auxiliary) data such as sequence-based predictions
and Gene Ontology information. 

** Results and conclusions

We find that integrating data, from a second more plentiful auxiliary
data source, directly in to classifier training and classifier
creation results in the assignment of proteins to organelles with high
generalisation accuracy. Furthermore, we show our hybrid, data fusion
approach outperforms a single classifier trained on each single data
source alone.

** Conclusion 

A new transfer learning framework for predicting protein localisation
to unify multiple sources of information to improve experimental
proteomics data is proposed. This novel machine learning methodology
forms part of the Bioconductor pRoloc [3] suite of computational 
methods available for organelle proteomics data analysis.

References:
[1] Gatto, L. et al., Proteomics 2010, 10, 3957
[2] Wu, P. and Dietterich, TG., ICML. 2004, 871.
[3] Gatto, L. et al. Bioinformatics 2014, In press.


NB: It is important to stress out in the abstract and poster/paper
that our main goal is to resolve localisation for a particular data
set, that the experimental data is what we care about and to
distinguish our work from the dry sub cellular folks.
