%\VignetteEngine{knitr}
%\VignetteIndexEntry{HUPO2014 RBioc poster}
%\VignetteKeywords{bioinformatics, proteomics, mass spectrometry}

\documentclass[final]{beamer}

\mode<presentation> {  %% check http://www-i6.informatik.rwth-aachen.de/~dreuw/latexbeamerposter.php for examples
  %% \usetheme{CCP}
}


% footline colors and fonts
\setbeamercolor{footline}{fg=white,bg=gray}
\setbeamerfont{footline}{fg=black,size=\normalsize}

\setbeamertemplate{footline}{
  \begin{beamercolorbox}[wd=\paperwidth]{upper separation line foot}
    \rule{0pt}{2pt}
  \end{beamercolorbox}
  \leavevmode%
  \begin{beamercolorbox}[ht=2ex]{footline}%
    \centering
    HUPO meeting
    \hskip3ex 
    5 -- 8 October 2014
    \hskip3ex 
    Madrid
  \end{beamercolorbox}
  \vskip0pt%
  \begin{beamercolorbox}[wd=\paperwidth]{lower separation line foot}
    \rule{0pt}{2pt}
  \end{beamercolorbox}
}


% \setbeamertemplate{bibliography item}[text]

\boldmath
\usepackage[orientation=portrait,size=a0,scale=1.4,debug]{beamerposter}                        % e.g. for DIN-A0 poster
%\usepackage[orientation=portrait,size=a1,scale=1.4,grid,debug]{beamerposter}                  % e.g. for DIN-A1 poster, with optional grid and debug output
%\usepackage[size=custom,width=200,height=120,scale=2,debug]{beamerposter}                     % e.g. for custom size poster
%\usepackage[orientation=portrait,size=a0,scale=1.0,printer=rwth-glossy-uv.df]{beamerposter}   % e.g. for DIN-A0 poster with rwth-glossy-uv printer check
% ...
%

%% hide navigation symbols (bottom right)
\setbeamertemplate{navigation symbols}{}
\setbeamertemplate{caption}[numbered]

\usepackage{ragged2e}
\usepackage[english]{babel}
\usepackage[latin1]{inputenc}
\usepackage{amsmath,amsthm, amssymb, latexsym}
\usefonttheme[onlymath]{serif}

\usepackage{graphicx}
\usepackage{caption}
\usepackage{subcaption}

\usepackage{tcolorbox}
\usepackage{changepage} %% provided adjustwidth
\usepackage{framed}

\newcommand{\R}{\texttt{R} }
\newcommand{\code}[1]{{\texttt{#1}}}
\newcommand{\Rfunction}[1]{{\texttt{#1}}}
\newcommand{\Robject}[1]{{\texttt{#1}}}
\newcommand{\Rpackage}[1]{{\mbox{\texttt{#1}}}}
\newcommand{\email}[1]{\href{mailto:#1}{\normalfont\texttt{#1}}}

\newcommand{\challenge}[1]{
       \begin{tcolorbox}[notitle,boxrule=1pt,colback=blue!10,colframe=blue!25]
         {#1}
       \end{tcolorbox}
}

\newcommand{\secintro}[1]{
  \bigskip
  \begin{tcolorbox}[notitle,boxrule=0pt,colback=blue!10,colframe=blue!10]{#1}\end{tcolorbox}}


%% colors
\definecolor{Red}{rgb}{0.7,0,0}
\definecolor{Blue}{rgb}{0,0,0.8}


\usepackage[bordercolor=white, backgroundcolor=gray!20]{todonotes}

\usepackage{hyperref}
\usepackage{breakurl}
\hypersetup{%
  pdfauthor={Laurent Gatto},%
  pdfusetitle,
  bookmarks = {true},
  bookmarksnumbered = {true},
  bookmarksopen = {true},
  bookmarksopenlevel = 2,
  unicode = {true},
  breaklinks = {false},
  hyperindex = {true},
  colorlinks = {true},
  linktocpage = {true},
  plainpages = {false},
  linkcolor = {Blue},
  citecolor = {Blue},
  urlcolor = {Red},
  pdfstartview = {Fit},
  pdfpagemode = {UseOutlines},
  pdfview = {XYZ null null null}
}



% %% figure numbering
% \usecaptiontemplate{
%   \small
%   \structure{\insertcaptionname~\insertcaptionnumber:}
%   \insertcaption
% }

\title[HUPO14 RBioc]{\huge A current perspective on using \R and Bioconductor for proteomics data analysis}

\author[Gatto et al.]{\large
  S. Gibb$^{1, 3}$,
  LM. Breckels$^{1,2}$,
  T. Lin Pedersen$^4$,
  VA. Petyuk$^5$,
  KS. Lilley$^2$ and
  L. Gatto$^{1,2,*}$}

\institute[CCP]{
  \begin{small}
    $^1$Computational Proteomics Unit and $^2$Cambridge Centre for Proteomics, Department of Biochemistry, Tennis Court Road, University of Cambridge, CB2 1QR\\
    $^3$Department of Anesthesiology and Intensive Care, Medical Faculty Carl Gustav Carus, Technical University Dresden, Fetscherstr. 74, 01307 Dresden\\
    $^4$Chr. Hansen A/S, H\o rsholm, Denmark / Technical University of Denmark, Kgs. Lyngby, Denmark\\
    $^5$Biological Sciences Division, Pacific Northwest National Laboratory, Richland, WA, USA\\
    \bigskip
    $^{*}$\url{lg390@cam.ac.uk} \hspace{5cm}  \url{http://cpu.sysbiol.cam.ac.uk}
  \end{small}
}

\date[]{}

\begin{document}

<<env, include=FALSE, echo=FALSE, cache=FALSE>>=
library("knitr")
opts_chunk$set(fig.align = 'center',
               fig.show = 'hold',
               par = TRUE,
               prompt = FALSE,
               eval = TRUE,
               stop_on_error = 1L,
               tidy = TRUE,
               comment = ':')
options(replace.assign = TRUE,
        width = 50)

## set.seed(1)
## suppressPackageStartupMessages(library("MSnbase"))
## suppressPackageStartupMessages(library("mzR"))
## suppressPackageStartupMessages(library("mzID"))
suppressPackageStartupMessages(library("MSnID"))
@
%%$

<<id0, cache=TRUE, echo=FALSE>>=
load("../R/msnid.rda")
id <- id0
id <- assess_termini(id, validCleavagePattern="[KR]\\.[^P]")
id <- assess_missed_cleavages(id, missedCleavagePattern="[KR](?=[^P$])")
id$msmsScore <- -log10(id$`MS-GF:SpecEValue`)
id$absParentMassErrorPPM <- abs(mass_measurement_error(id))
filter <- psmfilter
@

\begin{frame}[fragile]

  \maketitle

  \begin{columns}
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %% first column %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \begin{column}{.48\textwidth}
     \begin{block}{Introduction}
       \justifying \secintro{ The \R statistical environment and
         programming language is a key player in many domains that
         require robust data analysis. The Bioconductor project offer
         a wide range of \R packages dedicated to the analysis and
         comprehension of high throughput biology. Originally focused
         on genomics, R/Bioconductor are gaining increasing attention
         in the proteomics, metabolomics and mass spectrometry
         communities, as reflected by the download statistics and
         package contributions.}

      \begin{figure}[h]
        \begin{subfigure}[b]{0.4\linewidth}
          \includegraphics[width=\linewidth]{figures/development_biocviews.pdf}
        \end{subfigure}
        \begin{subfigure}[b]{0.4\linewidth}
          \includegraphics[width=\linewidth]{figures/downloads_biocviews.pdf}
        \end{subfigure}
        \label{fig:biocviews}
        \caption{\justifying The left figure shows the number of
          Bioconductor packages dedicated to \emph{Proteomics},
          \emph{MassSpectrometry} and \emph{MassSpectrometryData}
          packages on Bioconductor in the BiocViews . On the right
          figure, we show the number of distinct package
          downloads. Note that the current version, 2.14 currently
          encounters downloads. (NB: the data for Bioc 2.12 are
          interpolated due to massive scripted downloads).}
      \end{figure}
      
      Here, we present an overview of current Bioconductor
      infrastructure dedicated to proteomics and mass
      spectrometry. These software packages \textbf{tightly
        interoperate}, providing \textbf{well defined pipeline
        workflows} and a \textbf{flexible and in-depth development
        environment}.

    \end{block}

    \vspace{1cm} 

    \begin{block}{Working with raw data}
       \justifying
       \secintro{The proteomics community has developed a range of data
       standards and formats for MS data (e.g. \texttt{mzML}, \texttt{mzIdentML})
       to overcome the shortcomings of closed, binary vendor-specific formats.}

     One of the main projects that implement parsers for the XML-based
     open formats is the \texttt{C++} proteowizard project, which is
     interfaced by the \Rpackage{mzR} Bioconductor package using the
     \Rpackage{Rcpp} infrastructure for fast raw and (starting with
     Bioconductor version 3.0) identification data. \texttt{mzIdentML}
     files can also be parsed with the \Rpackage{mzID} package.

<<mzR, eval=FALSE>>=
library("mzR")
ms <- openMSfile("raw_data.raw")
id <- openIDfile("msgf-res.mzid")
library("mzID")
id2 <- mzID("msgf-res2.mzid")
@

       The resulting \Robject{ms} object is a file handle that allows fast
       access to the individual spectra. \Rpackage{mzR} is used by a
       variety of other packages like \Rpackage{xcms}, \Rpackage{MSnbase},
       \Rpackage{RMassBank} and \Rpackage{TargetSearch}.
     \end{block}
     \bigskip

     \begin{block}{Identification}
       \justifying \secintro{\texttt{R}/Bioconductor provide software
         to parse \texttt{mzIdentML} files (\Rpackage{mzID} and
         \Rpackage{mzR}, see above), directly run identification \R
         (\Rpackage{rTANDEM}, \Rpackage{MSGFplus}) and optimise FDR
         calculations based on decoy searches. Below, we use
         \Rpackage{MSnID} to define identification searches filters
         using precursor mass error, identification scores and missed
         cleavage to minimise false discovery rates. }

         <<ident, cache=TRUE>>=
         (id <- apply_filter(id, filter))
         @
     \end{block}

      \begin{block}{Quantitation}
        \justifying \secintro{Several quantitation pipelines are
          supported: \Rpackage{MSnbase} and \Rpackage{isobar} for
          isobaric tagging such as iTRAQ and TMT,
          \Rpackage{MALDIquant} for MALDI data, \Rpackage{MSnbase} for
          spectra counting and other MS$^2$ label free methods and
          \Rpackage{synapter} supports a DIA complete pipeline,
           including ion mobility separation on Waters Synapt
          instruments.}
      \end{block}

    \end{column}

    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %% second column %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \begin{column}{.48\textwidth}

      \begin{block}{MS data processing}
        \secintro{MS spectrum processing is available in multiple
          packages, optimised for specific use cases and pipelines:
          \Rpackage{MSnbase}, \Rpackage{xcms}, \Rpackage{MALDIquant}.}

      \begin{figure}[h]
          \includegraphics[width=1\linewidth]{figures/mqpreproc.pdf}
       \label{fig:msproc}
       \caption{\justifying Illustration of the \texttt{MALDIquant}
         preprocessing pipeline. The first figure (A) shows a raw
         spectrum with the estimated baseline. In the second figure
         (B) the spectrum is variance-stabilised, smoothed,
         baseline-corrected and the detected peaks are marked with
         blue crosses. The third figure (C) is an example of a fitted
         warping function for peak alignment. In the next figures,
         four peaks are shown before (D) and after (E) performing the
         alignment. The last figure (F) represents a merged spectrum
         with discovered and labelled peaks.}
      \end{figure}
        
      \end{block}

      \begin{block}{Visualisation}
        \justifying \secintro{ Using \R for high quality programmable
          figures and interactive graphical user interfaces such as
          \url{https://lgatto.shinyapps.io/shinyMA/}. }

        \begin{columns}
          \begin{column}{.4\textwidth}
            \begin{figure}[h]
              \includegraphics[width = .3\linewidth]{figures/mapsheadpdf.pdf}
              \includegraphics[width = .3\linewidth]{figures/maps3Dpdf.pdf}
              \includegraphics[width = .3\linewidth]{figures/maps3D2pdf.pdf} \\
              \includegraphics[width = .9\linewidth]{figures/msdetailspdf.pdf}
              \label{fig:vis2}
            \end{figure}            
          \end{column}
          \begin{column}{.4\textwidth}
            \begin{figure}[h]
              \includegraphics[width=\linewidth]{figures/sweave-precplot.pdf}
              \label{fig:vis1}
            \end{figure}            
          \end{column}
        \end{columns}

        \begin{figure}[h]
          \includegraphics[width=.4\linewidth]{figures/gui1.png}\hspace{2cm}
          \includegraphics[width=.4\linewidth]{figures/gui2.png}
          \caption{Visualising raw data and process quantitative data
            (top) and interactive graphical interfaces (bottom) from
            the \Rpackage{pRolocGUI} and \Rpackage{MSGFgui} packages. }
          \label{fig:gui}
        \end{figure}            
        
      \end{block}

      \vfill

      \begin{block}{Data processing, statistics and machine learning}
        \justifying

        \secintro{As an environment for statistical computing, the
          very best of data processing, statistical modelling and
          machine learning is readily available in packages such as
          \Rpackage{MSnbase}, \Rpackage{isobar}, \Rpackage{MSstats},
          \Rpackage{msmsTests}.}

        \Rpackage{MSstats} offers set of tools for statistical
        relative protein significance analysis in DDA, SRM and DIA
        experiments as well as power calculation. \Rpackage{msmsTests}
        leverages powerful statistical modelling from the
        \Rpackage{edgeR} package to analyse spectral counting data. 

      \end{block}

      \vspace{1cm}

      \begin{block}{Conclusion}
        \secintro{A wide range of other applications such as
          post-translational modifications (\Rpackage{isobar}),
          quality control (\Rpackage{qcmetrics} and
          \Rpackage{proteoQC}) or spatial proteomics
          (\Rpackage{pRoloc}) are available as Bioconductor
          packages. A complete list is available on the Bioconductor
          page and in the \Rpackage{RforProteomics} package.
          Contributions and discussions are welcome on our Google
          group and GitHub wiki.  }
      \end{block}

      \begin{columns}
        \begin{column}{.75\textwidth}
          \begin{description}
          \item[The R project] \url{http://www.r-project.org/}
          \item[Bioconductor] \url{http://bioconductor.org/}
          \item[RforProteomics]\url{http://is.gd/R4Proteomics}
          \item[Google group] \url{https://is.gd/rbioc_sig_proteomics}
          \item[Support site] \url{https://support.bioconductor.org}
          \end{description}              
        \end{column}
        \begin{column}{.23\textwidth}
          \includegraphics[width=.9\linewidth]{./figures/RBioc_QA.png}
        \end{column}
      \end{columns}

      \vspace{1cm}

      {\small This work was supported by the \textbf{European Union
          7$^{th}$ Framework Program PRIME-XS project} and a
        \textbf{BBSRC Tools and Resources Development Fund}.}

      \vfill

    \end{column}
  \end{columns}

\end{frame}


\end{document}

