%\VignetteEngine{knitr}
%\VignetteIndexEntry{HUPO2014 RBioc poster}
%\VignetteKeywords{bioinformatics, proteomics, mass spectrometry}

\documentclass[final]{beamer}

\mode<presentation> {  %% check http://www-i6.informatik.rwth-aachen.de/~dreuw/latexbeamerposter.php for examples
  %% \usetheme{CCP}
}

\setbeamertemplate{bibliography item}[text]

\boldmath
\usepackage[orientation=portrait,size=a0,scale=1.4,debug]{beamerposter}                        % e.g. for DIN-A0 poster
%\usepackage[orientation=portrait,size=a1,scale=1.4,grid,debug]{beamerposter}                  % e.g. for DIN-A1 poster, with optional grid and debug output
%\usepackage[size=custom,width=200,height=120,scale=2,debug]{beamerposter}                     % e.g. for custom size poster
%\usepackage[orientation=portrait,size=a0,scale=1.0,printer=rwth-glossy-uv.df]{beamerposter}   % e.g. for DIN-A0 poster with rwth-glossy-uv printer check
% ...
%

%% hide navigation symbols (bottom right)
\setbeamertemplate{navigation symbols}{}

\usepackage{ragged2e}
\usepackage[english]{babel}
\usepackage[latin1]{inputenc}
\usepackage{amsmath,amsthm, amssymb, latexsym}
\usefonttheme[onlymath]{serif}

\usepackage{graphicx}
\usepackage{caption}
\usepackage{subcaption}

\usepackage{tcolorbox}
\usepackage{changepage} %% provided adjustwidth
\usepackage{framed}

\newcommand{\R}{\texttt{R} }
\newcommand{\code}[1]{{\texttt{#1}}}
\newcommand{\Rfunction}[1]{{\texttt{#1}}}
\newcommand{\Robject}[1]{{\texttt{#1}}}
\newcommand{\Rpackage}[1]{{\mbox{\texttt{#1}}}}
\newcommand{\email}[1]{\href{mailto:#1}{\normalfont\texttt{#1}}}

\newcommand{\challenge}[1]{
       \begin{tcolorbox}[notitle,boxrule=1pt,colback=blue!10,colframe=blue!25]
         {#1}
       \end{tcolorbox}
}

\newcommand{\secintro}[1]{
  \bigskip
  \begin{tcolorbox}[notitle,boxrule=0pt,colback=blue!10,colframe=blue!10]{#1}\end{tcolorbox}}


%% colors
\definecolor{Red}{rgb}{0.7,0,0}
\definecolor{Blue}{rgb}{0,0,0.8}


\usepackage[bordercolor=white, backgroundcolor=gray!20]{todonotes}

\usepackage{hyperref}
\usepackage{breakurl}
\hypersetup{%
  pdfauthor={Laurent Gatto},%
  pdfusetitle,
  bookmarks = {true},
  bookmarksnumbered = {true},
  bookmarksopen = {true},
  bookmarksopenlevel = 2,
  unicode = {true},
  breaklinks = {false},
  hyperindex = {true},
  colorlinks = {true},
  linktocpage = {true},
  plainpages = {false},
  linkcolor = {Blue},
  citecolor = {Blue},
  urlcolor = {Red},
  pdfstartview = {Fit},
  pdfpagemode = {UseOutlines},
  pdfview = {XYZ null null null}
}



%% figure numbering
\usecaptiontemplate{
  \small
  \structure{\insertcaptionname~\insertcaptionnumber:}
  \insertcaption
}

\title[HUPO14 RBioc]{\huge A current perspective on using \R and Bioconductor for proteomics data analysis}

\author[Gatto et al.]{\large L. Gatto$^{*, 1,2}$, S. Gibb$^{1,3}$}

\institute[CCP]{
  \begin{small}
    $^1$Computational Proteomics Unit and $^2$Cambridge Centre for Proteomics, Department of Biochemistry, Tennis Court Road, University of Cambridge, CB2 1QR\\
    $^3$Department of Anesthesiology and Intensive Care, Medical Faculty Carl Gustav Carus, Technical University Dresden, Fetscherstr. 74, 01307 Dresden\\
    \bigskip
    $^{*}$\url{lg390@cam.ac.uk} -- \url{http://www.bio.cam.ac.uk/proteomics/}
  \end{small}
}

\date[]{}

\begin{document}

<<env, include=FALSE, echo=FALSE, cache=FALSE>>=
library("knitr")
opts_chunk$set(fig.align = 'center',
               fig.show = 'hold',
               par = TRUE,
               prompt = FALSE,
               eval = TRUE,
               stop_on_error = 1L,
               tidy = TRUE,
               comment = ':')
options(replace.assign = TRUE,
        width = 50)

## set.seed(1)
## suppressPackageStartupMessages(library("MSnbase"))
## suppressPackageStartupMessages(library("mzR"))
## suppressPackageStartupMessages(library("mzID"))
@
%%$

\begin{frame}[fragile]

  \maketitle

  \begin{columns}
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %% first column %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \begin{column}{.48\textwidth}
     \begin{block}{Introduction}
       \justifying

       The \R statistical environment and programming language is a
       key player in many domains that require robust data
       analysis. The Bioconductor project offer a wide range of \R
       packages dedicated to the analysis and comprehension of high
       throughput biology. Originally focused on genomics,
       R/Bioconductor are gaining increasing attention in the
       proteomics, metabolomics and mass spectrometry communities, as
       reflected by the download statistics and package contributions.

      \begin{figure}[h]
        \begin{subfigure}[b]{0.44\linewidth}
          \includegraphics[width=\linewidth]{figures/development_biocviews.pdf}
        \end{subfigure}
        \begin{subfigure}[b]{0.44\linewidth}
          \includegraphics[width=\linewidth]{figures/downloads_biocviews.pdf}
        \end{subfigure}
        \label{fig:biocviews}
        \caption{The left figure shows the number of Bioconductor
          packages dedicated to \emph{Proteomics},
          \emph{MassSpectrometry} and \emph{MassSpectrometryData}
          packages on Bioconductor in the BiocViews . On the right
          figure, we show the number of distinct package
          downloads. Note that the current version, 2.14 currently
          encounters downloads. (NB: the data for Bioc 2.12 are
          interpolated due to massive scripted downloads).}
      \end{figure}
      
      Here, we present an overview of current Bioconductor
      infrastructure dedicated to proteomics and mass spectrometry.

     \end{block}
     \vfill

    \begin{block}{Working with raw data}
       \justifying
       \secintro{The proteomics community has developed a range of data
       standards and formats for MS data (e.g. \texttt{mzML}, \texttt{mzIdentML})
       to overcome the shortcomings of closed, binary vendor-specific formats.}

     One of the main projects that implement parsers for the XML-based
     open formats is the \texttt{C++} proteowizard project (Chambers
     \textit{et al.} Nat Biotechnol. 2012 Oct;30(10):918-20), which is
     interfaced by the \Rpackage{mzR} Bioconductor package using the
     \Rpackage{Rcpp} infrastructure for fast raw and (starting with
     Bioconductor version 3.0) identification data. \texttt{mzIdentML}
     files can also be parsed with the \Rpackage{mzID} package.

<<mzR, eval=FALSE>>=
library("mzR")
ms <- openMSfile("raw_data.raw")
id <- openIDfile("msgf-res.mzid")
library("mzID")
id2 <- openIDfile("msgf-res2.mzid")
@

       The resulting \Robject{ms} object is a file handle that allows fast
       random access to the individual spectra. \Rpackage{mzR} is used by a
       variety of other packages like \Rpackage{xcms}, \Rpackage{MSnbase},
       \Rpackage{RMassBank} and \Rpackage{TargetSearch}.
     \end{block}
     \bigskip

     \begin{block}{Identification}
       \justifying \secintro{\R/Bioconductor provide software to parse
         \texttt{mzIdentML} files (\Rpackage{mzID} and
         \Rpackage{mzR}, see above), directly run identification \R
         (\Rpackage{rTANDEM}, \Rpackage{MSGFplus}) and optimise FDR
         calculations based on decoy searches (\Rpackage{MSnID})}
<<ident>>=
## an MSnID chunk
@

     \end{block}

      \begin{block}{MS data processing}
        \secintro{MS spectrum processing is available in multiple
          packages, optimised for specific use cases and pipelines:
          \Rpackage{MSnbase}, \Rpackage{xcms}, \Rpackage{MALDIquant}.}

      \begin{figure}[h]
          \includegraphics[width=.9\linewidth]{figures/mqpreproc.pdf}
        \label{fig:msproc}
        \caption{Illustration of the \texttt{MALDIquant} preprocessing
          pipeline. The first figure (A) shows a raw spectrum with the
          estimated baseline. In the second figure (B) the spectrum is
          variance-stabilised, smoothed, baseline-corrected and the
          detected peaks are marked with blue crosses. The third
          figure (C) is an example of a fitted warping function for
          peak alignment. In the next figures, four peaks are shown
          before (D) and after (E) performing the alignment. The last
          figure (F) represents a merged spectrum with discovered and
          labelled peaks.}
      \end{figure}
        
      \end{block}

    \end{column}

    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %% second column %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \begin{column}{.48\textwidth}

      \begin{block}{Quantitation}
        \justifying \secintro{Several quantitation pipelines are
          supported: \Rpackage{MSnbase} and \Rpackage{isobar} for
          isobaric tagging such as iTRAQ and TMT,
          \Rpackage{MALDIquant} for MALDI data, \Rpackage{MSnbase} for
          spectra couting and other MS$^2$ label free methods and
          \Rpackage{synapter} supports a DIA complete pipeline,
           including ion mobility separation on Waters Synapt
          instruments.}
      \end{block}

      \begin{block}{Visualisation}
        \justifying \secintro{\R excels in graphics. It also allows to
          easily program interactive visualisation interfaces.}
        A figure from the vis paper, a \Rpackage{shiny} GUI example.

      \end{block}

      \vfill

      \begin{block}{Data processing, statistics and machine learning}
        \justifying

        \secintro{As an environment for statistical computing, the
          very best of data processing, statistical modelling and
          machine learning is readily available.}  

        \Rpackage{MSnbase}, \Rpackage{isobar}, \Rpackage{MSstats},
        \Rpackage{msmsTests},

      \end{block}

      \vfill

      \begin{block}{Misc}
        Any other applications: PTMs, spatial proteomics, ...
      \end{block}

      \begin{block}{}
        \challenge{

        \begin{thebibliography}{6}

        % \bibitem{msstats} Choi M \textit{et al.}
        %   \emph{MSstats: an R package for statistical analysis of quantitative
        %   mass spectrometry-based proteomic experiments}.
        %   Bioinformatics. 2014 %% doi:10.1093/bioinformatics/btu305
        %   PMID: 24794931.


        \bibitem{r4p} Gatto L and Christoforou A
          \emph{Using R and Bioconductor for proteomics data analysis}.
          Biochim Biophys Acta. 2013 %% May 18. doi:pii: S1570-9639(13)00186-6.10.1016/j.bbapap.2013.04.032.
          PMID: 23692960.

        % \bibitem{msnbase} Gatto L and Lilley KS
        %   \emph{MSnbase-an R/Bioconductor package for isobaric tagged mass spectrometry data visualization, processing and quantitation}.
        %   Bioinformatics. 2012 %% Jan; 28:(2)288-9
        %   PMID: 22113085.

        % \bibitem{maldiquant} Gibb S and Strimmer K
        %   \emph{MALDIquant: a versatile R package for the analysis of mass spectrometry data.}
        %   Bioinformatics. 2012 %% Sep; 28:(17)2270-1
        %   PMID: 22796955.

          \begin{description}
          \item[R] \url{http://www.r-project.org/}
          \item[Bioconductor] \url{http://bioconductor.org/}
          \item[RforProteomics]\url{http://is.gd/R4Proteomics}
          \item[Google group] \url{https://is.gd/rbioc_sig_proteomics}
          \item[Support site] \url{https://support.bioconductor.org}
          \end{description}
          
        \end{thebibliography}
      }

      \end{block}

      %% \vspace{2mm}
    \end{column}
  \end{columns}

\end{frame}


\end{document}

