%\VignetteEngine{knitr}
%\VignetteIndexEntry{HUPO2014 RBioc poster}
%\VignetteKeywords{bioinformatics, proteomics, mass spectrometry}

\documentclass[final]{beamer}

\mode<presentation> {  %% check http://www-i6.informatik.rwth-aachen.de/~dreuw/latexbeamerposter.php for examples
  %% \usetheme{CCP}
}

\setbeamertemplate{bibliography item}[text]

\boldmath
\usepackage[orientation=portrait,size=a0,scale=1.4,debug]{beamerposter}                        % e.g. for DIN-A0 poster
%\usepackage[orientation=portrait,size=a1,scale=1.4,grid,debug]{beamerposter}                  % e.g. for DIN-A1 poster, with optional grid and debug output
%\usepackage[size=custom,width=200,height=120,scale=2,debug]{beamerposter}                     % e.g. for custom size poster
%\usepackage[orientation=portrait,size=a0,scale=1.0,printer=rwth-glossy-uv.df]{beamerposter}   % e.g. for DIN-A0 poster with rwth-glossy-uv printer check
% ...
%

%% hide navigation symbols (bottom right)
\setbeamertemplate{navigation symbols}{}

\usepackage{ragged2e}
\usepackage[english]{babel}
\usepackage[latin1]{inputenc}
\usepackage{amsmath,amsthm, amssymb, latexsym}
\usefonttheme[onlymath]{serif}

\usepackage{graphicx}
\usepackage{caption}
\usepackage{subcaption}

\usepackage{tcolorbox}
\usepackage{changepage} %% provided adjustwidth
\usepackage{framed}

\newcommand{\R}{\texttt{R} }
\newcommand{\code}[1]{{\texttt{#1}}}
\newcommand{\Rfunction}[1]{{\texttt{#1}}}
\newcommand{\Robject}[1]{{\texttt{#1}}}
\newcommand{\Rpackage}[1]{{\mbox{\texttt{#1}}}}
\newcommand{\email}[1]{\href{mailto:#1}{\normalfont\texttt{#1}}}

\newcommand{\challenge}[1]{
       \begin{tcolorbox}[notitle,boxrule=1pt,colback=blue!10,colframe=blue!25]
         \textbf{Challenges}~{#1}
       \end{tcolorbox}
}

\newcommand{\secintro}[1]{
  \bigskip
  \begin{tcolorbox}[notitle,boxrule=0pt,colback=blue!10,colframe=blue!10]{#1}\end{tcolorbox}}


%% colors
\definecolor{Red}{rgb}{0.7,0,0}
\definecolor{Blue}{rgb}{0,0,0.8}


\usepackage[bordercolor=white, backgroundcolor=gray!20]{todonotes}

\usepackage{hyperref}
\usepackage{breakurl}
\hypersetup{%
  pdfauthor={Laurent Gatto},%
  pdfusetitle,
  bookmarks = {true},
  bookmarksnumbered = {true},
  bookmarksopen = {true},
  bookmarksopenlevel = 2,
  unicode = {true},
  breaklinks = {false},
  hyperindex = {true},
  colorlinks = {true},
  linktocpage = {true},
  plainpages = {false},
  linkcolor = {Blue},
  citecolor = {Blue},
  urlcolor = {Red},
  pdfstartview = {Fit},
  pdfpagemode = {UseOutlines},
  pdfview = {XYZ null null null}
}



%% figure numbering
\usecaptiontemplate{
  \small
  \structure{\insertcaptionname~\insertcaptionnumber:}
  \insertcaption
}

\title[HUPO14 RBioc]{\huge A current perspective on using \R and Bioconductor for proteomics data analysis}

\author[Gatto et al.]{\large L. Gatto$^{*, 1,2}$, S. Gibb$^{2,3}$}

\institute[CCP]{
  \begin{small}
    $^1$Cambridge Centre for Proteomics, Department of Biochemistry, Tennis Court Road, University of Cambridge, CB2 1QR\\
    $^2$Computational Proteomics Unit, Department of Biochemistry, Tennis Court Road, University of Cambridge, CB2 1QR\\
    $^3$Department of Anesthesiology and Intensive Care, Medical Faculty Carl Gustav Carus, Technical University Dresden, Fetscherstr. 74, 01307 Dresden\\
    \bigskip
    $^{*}$\url{lg390@cam.ac.uk} -- \url{http://www.bio.cam.ac.uk/proteomics/}
  \end{small}
}

\date[]{}

\begin{document}

<<env, include=FALSE, echo=FALSE, cache=FALSE>>=
library("knitr")
opts_chunk$set(fig.align = 'center',
               fig.show = 'hold',
               par = TRUE,
               prompt = FALSE,
               eval = TRUE,
               stop_on_error = 1L,
               tidy = TRUE,
               comment = ':')
options(replace.assign = TRUE,
        width = 50)

set.seed(1)

suppressPackageStartupMessages(library("MSnbase"))
suppressPackageStartupMessages(library("mzR"))
suppressPackageStartupMessages(library("mzID"))
@
%%$

\begin{frame}[fragile]

  \maketitle

  \begin{columns}
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %% first column %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \begin{column}{.48\textwidth}
     \begin{block}{Introduction}
       \justifying
     \end{block}

     \begin{block}{Bioconductor package submissions and downloads}
      \begin{figure}[h]
        \begin{subfigure}[b]{0.49\linewidth}
          \includegraphics[width=\linewidth]{figures/development_biocviews_sep2014}
        \end{subfigure}
        \begin{subfigure}[b]{0.49\linewidth}
          \includegraphics[width=\linewidth]{figures/downloads_biocviews_sep2014}
        \end{subfigure}
        \label{fig:biocviews}
        \caption{The left figure shows the growing number of packages on
        Bioconductor in the BiocViews \emph{Proteomics, MassSpectrometry} and
        \emph{MassSpectrometryData}. On the right figure the number of distinct
        package downloads in these BiocViews are depicted (please note that
        the data for Bioc 2.12 are interpolated because of server problems in
        this time period).}
      \end{figure}

     \end{block}
     \vfill

    \begin{block}{Working with raw data}
       \justifying
       \secintro{The proteomics community has developed a range of data
       standards and formats for MS data (e.g. \texttt{mzML}, \texttt{mzIdentML})
       to overcome the shortcomings of closed, binary vendor-specific formats.}

       One of the main projects that implement parsers for the XML-based
       open formats is the \texttt{C++} proteowizard project
       \cite{Chambers2012}, which is interfaced by the
       \Rpackage{mzR} Bioconductor package using the \Rpackage{Rcpp}
       infrastructure.

<<mzR>>=
library("mzR")
fname <- dir(system.file(package="MSnbase", dir="extdata"), full.name=TRUE, pattern="mzXML$")
ms <- openMSfile(fname)
@

       The resulting \Robject{ms} object is a file handle that allows fast
       random access to the individual spectra. \Rpackage{mzR} is used by a
       variety of other packages like \Rpackage{xcms}, \Rpackage{MSnbase},
       \Rpackage{RMassBank} and \Rpackage{TargetSearch}.
     \end{block}
     \bigskip

     \begin{block}{Labelled quantitation}
       \justifying
       \secintro{The same raw data file can be imported in a convenient higher level container and
       directly processed, plotted, quantified and normalised with the \Rpackage{MSnbase} \cite{msnbase} software.}

<<msnbase, eval = FALSE>>=
exp <- readMSData(fname, verbose = FALSE)
plot(exp[["X3.1"]], full = TRUE, reporters = iTRAQ4)
set <- quantify(exp, method = "trap", reporters = iTRAQ4, verbose = FALSE, parallel = TRUE)
head(exprs(set), n = 3)
@

      \begin{figure}[h]
        \centering
        \begin{subfigure}[b]{0.55\linewidth}
          \centering
          \includegraphics[width=\linewidth]{figures/plotspectrum}
        \end{subfigure}
        \begin{subfigure}[b]{0.44\linewidth}
          \includegraphics[width=\linewidth]{figures/spikes} \\
          \includegraphics[width=\linewidth]{figures/plotmzdelta}
        \end{subfigure}
        \label{fig:msnbase}
        \caption{MS$^2$ spectrum of an iTRAQ 4-plex experiment highlighting the 4 isobaric reporter ions,
          as produced by \Rfunction{plot} above (left). Peptides of interest from a spiked-in experiment
          (top right) and distribution of the \textit{m/z} differences of all MS$^2$ spectra from the same
          experiment, used as a peptide-spectrum matching quality assessment (bottom right).}
      \end{figure}
      \bigskip

     \end{block}


    \end{column}



    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %% second column %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \begin{column}{.48\textwidth}

      \begin{block}{Label-free quantitation}
        \justifying
        \secintro{Support for data dependent label-free quantitation is available, among others, in the
        \Rpackage{MSnbase} \cite{msnbase}, \Rpackage{xcms} \cite{xcms} and \Rpackage{MALDIquant} \cite{maldiquant} packages.}
        %% \begin{Leftbar}
          The latter provides a complete pipeline, including
          baseline subtraction, smoothing, peak detection and alignment using warping functions,
          handling of replicated measurements as well as allowing spectra with different resolutions.
          \begin{figure}[h]
            %% please find the script to create this figure at
            %% https://github.com/sgibb/MALDIquantExamples/blob/master/createFigure1_color.R
            \includegraphics[width=\textwidth]{figures/mqplot2}
            \label{fig:mq}
            \caption{Illustration of the \Rpackage{MALDIquant} pipeline:
              raw spectrum with estimate baseline (A);
              variance-stabilised, smoothed, baseline-corrected spectrum with detected peaks (B);
              fitted warping function for peak alignment (C);
              four unaligned peaks (D);
              four aligned peaks (E);
              merged spectrum with discovered and labelled peaks (F).}
          \end{figure}

          A complete pipeline for MS$^E$ data independent acquisition, including support for ion mobility separation
          is available in the \Rpackage{synapter} package \cite{synapter} that, among other features,
          transfers identification between acquisitions to substantially reduce missing values.
        \bigskip
      \end{block}

      \begin{block}{Peptide identification}
        \justifying
        \secintro{The recently released \Rpackage{rTANDEM} package encapsulates
        the \texttt{X!Tandem} \cite{tandem} search engine in \texttt{R}.}
          It uses the same XML-based parameter files as the native application
          or dedicated \R parameter object.
          Result files can be directly parsed and mined in \R.

<<rtandem, eval = FALSE, tidy = FALSE>>=
xmlres <- rtandem(spectra.mgf, taxon = "yeast",
                  taxonomy = "taxonomy.xml",
                  default.parameters = "default-params.xml")
## or xmlres <- tandem(param)
res <- GetResultsFromXML(xmlres)
proteins <- GetProteins(res) ## data.table objects
peptides <- GetPeptides(res)
@
          \secintro{
          The latest \texttt{mzR} and the \texttt{mzID} packages offer support
          for \texttt{mzIdentML} files and allow to import identification data
          from most commonly used search engines.}

<<mzid, eval = FALSE, tidy = FALSE>>=
library("mzR")
ident <- openIDfile("identification.mzid")
# or
library("mzID")
ident <- mzID("identification.mzid")
@

      \end{block}

      \vfill

      \begin{block}{Statistics}
        \justifying
        While there exists a lot of excellent packages for differential
        expression tests in \R \texttt{MSstats} and \texttt{msmsTests} offer
        tests that are well integrated into the \texttt{MSnbase} ecosystem.

<<msstats, eval = FALSE, tidy = FALSE>>=
## MSstats/msTests example?
@
      \end{block}

      \vfill

      \begin{block}{}
        \justifying
        \tiny

        \begin{thebibliography}{6}

        \bibitem{synapter} Bond NJ \textit{et al.}
          \emph{Improving Qualitative and Quantitative Performance for MS$^E$-based Label-free Proteomics}.
          J Proteome Res. 2013 %% Jun 7;12(6):2340-53. doi: 10.1021/pr300776t. Epub 2013 Apr 17.
          PMID: 23510225.

        \bibitem{Chambers2012} Chambers M. \textit{et al.}
          \emph{A cross-platform toolkit for mass spectrometry and proteomics}.
          Nat Biotechnol. 2012 %% Oct;30(10):918-20. doi: 10.1038/nbt.2377.
          PMID: 23051804.

        \bibitem{r4p} Gatto L and Christoforou A.
          \emph{Using R and Bioconductor for proteomics data analysis}.
          Biochim Biophys Acta. 2013 %% May 18. doi:pii: S1570-9639(13)00186-6.10.1016/j.bbapap.2013.04.032.
          PMID: 23692960.

        \bibitem{msnbase} Gatto L and Lilley KS
          \emph{MSnbase-an R/Bioconductor package for isobaric tagged mass spectrometry data visualization, processing and quantitation}.
          Bioinformatics. 2012 %% Jan; 28:(2)288-9
          PMID: 22113085.

        \bibitem{maldiquant} Gibb S and Strimmer K
          \emph{MALDIquant: a versatile R package for the analysis of mass spectrometry data.}
          Bioinformatics. 2012 %% Sep; 28:(17)2270-1
          PMID: 22796955.

        \bibitem{xcms} Smith CA \textit{et al.}
          \emph{XCMS: Processing mass spectrometry data for metabolite profiling using nonlinear peak alignment, matching and identification}.
          Analytical Chemistry 2006 %% 78:779-787
          PMID: 16448051.

        \end{thebibliography}
      \end{block}

      %% \vspace{2mm}
    \end{column}
  \end{columns}

\end{frame}


\end{document}

